### ISSUE TITLE
Skia->Engine roller blocked on displaylist failures

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#137558]
Example failure: https://github.com/flutter/engine/pull/47467

## log
```
FAILED TESTS (10/710):
      34 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpsMightSupportGroupOpacityBlendMode (try #1)
      38 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpsMightSupportGroupOpacityBlendMode (try #2)
      26 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpSizes (try #1)
      30 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpSizes (try #2)
      26 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListComplexity.DrawTextBlob (try #1)
      27 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListComplexity.DrawTextBlob (try #2)
      26 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpDisplayListsNotEqualEmpty (try #1)
      25 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpDisplayListsNotEqualEmpty (try #2)
      29 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpDisplayListsCompareToEachOther (try #2)
      34 ms: /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpDisplayListsCompareToEachOther (try #1)
SUMMARY:
  /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListComplexity.DrawTextBlob passed 0 / 2 times.
  /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpDisplayListsCompareToEachOther passed 0 / 2 times.
  /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpDisplayListsNotEqualEmpty passed 0 / 2 times.
  /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpSizes passed 0 / 2 times.
  /Volumes/Work/s/w/ir/cache/builder/src/out/host_debug/display_list_unittests DisplayListTest.SingleOpsMightSupportGroupOpacityBlendMode passed 0 / 2 times.
```

## example failure

```
[411/710] DisplayListTest.SingleOpDisplayListsCompareToEachOther (29 ms)
[INFO:test_timeout_listener.cc(76)] Test timeout of 300 seconds per test case will be enforced.
�[0;33mNote: Google Test filter = DisplayListTest.SingleOpDisplayListsCompareToEachOther
�[m�[0;32m[==========] �[mRunning 1 test from 1 test suite.
�[0;32m[----------] �[mGlobal test environment set-up.
�[0;32m[----------] �[m1 test from DisplayListTest
�[0;32m[ RUN      ] �[mDisplayListTest.SingleOpDisplayListsCompareToEachOther
../../flutter/display_list/display_list_unittests.cc:653: Failure
Value of: listA->Equals(*listB)
  Actual: true
Expected: false
DrawTextBlob(variant 1 ==? variant 2)
�[0;31m[  FAILED  ] �[mDisplayListTest.SingleOpDisplayListsCompareToEachOther (1 ms)
```

cc @flar

**Comment by gaaclarke** [comment]
The `SingleOpDisplayListsCompareToEachOther` hasn't changed in over a year: https://github.com/flutter/engine/blame/0b6d9e13f1b28b70dd94f18bb7f3aa2a14f6a387/display_list/display_list_unittests.cc#L625

**Comment by gaaclarke** [comment]
Here are the 2 commits in that roll: https://skia.googlesource.com/skia.git/+log/817fca468240..bca43ec489e5

One has to do with canvas kit, the other has to do with text.  It seems unlikely that those changes would cause failures.

**Comment by gaaclarke** [comment]
The whole display_list portion of the repository hasn't changed since oct 20: https://github.com/flutter/engine/commits/0b6d9e13f1b28b70dd94f18bb7f3aa2a14f6a387/display_list

**Comment by gaaclarke** [comment]
This failure sounds more linked to the text change:
```
[0;32m[ RUN      ] [mDisplayListTest.SingleOpsMightSupportGroupOpacityBlendMode
../../flutter/display_list/display_list_unittests.cc:1041: Failure
Expected equality of these values:
  display_list->can_apply_group_opacity()
    Which is: true
  expect_for_op
    Which is: false
{
  receiver.drawTextBlob(TestBlob1, 0, 0);
}
../../flutter/display_list/display_list_unittests.cc:1054: Failure
Expected equality of these values:
  display_list->can_apply_group_opacity()
    Which is: true
  expect_with_kSrc
    Which is: false
{
  receiver.setBlendMode(kSrc);
  receiver.drawTextBlob(TestBlob1, 0, 0);
}
[0;31m[  FAILED  ] [mDisplayListTest.SingleOpsMightSupportGroupOpacityBlendMode (0 ms)
```

**Comment by gaaclarke** [comment]
I was able to verify locally that bca43ec489e5fa23c08f598b5747a6e83ef04277 is causing the failures.

**Comment by gaaclarke** [comment]
I reached out to @kjlubick and he said that this is because flutter engine is using skia's `BUILD.gn` files directly.  Specifically he changed the `SK_DEFAULT_TYPEFACE_IS_EMPTY` for skia and surprisingly to him it changed flutter's behavior.

**Comment by kjlubick** [comment]
Fix incoming in https://skia-review.googlesource.com/c/skia/+/771703

**Comment by gaaclarke** [comment]
@flar made a good point offline that those tests were flimsy to be testing with a default font anyway and filed an issue to address that: https://github.com/flutter/flutter/issues/137565

**Comment by gaaclarke** [comment]
This isn't showing up anymore.  However there is another issue now manifesting.  I'll file a new issue.  Thanks for the help on this one everyone.
